poetry/core/packages/package.py

```diff
@@ -9,28 +8,20 @@
 from typing import Optional
 from typing import Union
 
-from poetry.core.semver import Version
-from poetry.core.semver import parse_constraint
-from poetry.core.spdx import License
-from poetry.core.spdx import license_by_id
-from poetry.core.version.markers import AnyMarker
+from poetry.core.semver.helpers import parse_constraint
 from poetry.core.version.markers import parse_marker
 
-# Do not move to the TYPE_CHECKING only section, because Dependency get's imported
-# by poetry/packages/locker.py from here
-from .dependency import Dependency
 from .specification import PackageSpecification
 from .utils.utils import create_nested_marker
 
 
 if TYPE_CHECKING:
-    from poetry.core.semver import VersionTypes  # noqa
+    from poetry.core.semver.helpers import VersionTypes  # noqa
+    from poetry.core.semver.version import Version  # noqa
+    from poetry.core.spdx.license import License  # noqa
     from poetry.core.version.markers import BaseMarker  # noqa
 
-    from .directory_dependency import DirectoryDependency
-    from .file_dependency import FileDependency
-    from .url_dependency import URLDependency
-    from .vcs_dependency import VCSDependency
+    from .types import DependencyTypes
```

Comment:
I still think we need to keep `from .dependency import Dependency` in this file outside the `TYPE_CHECKING` only section. Otherwise this this will break any poetry version <1.2 installed via pip(x).

Reply:
We will need to fix it on Poetry's side then, I think, rather than keeping it here.

Reply:
@finswimmer Also, the next bugfix release of Poetry should change the `poetry-core` constraint to `< 1.2` to ensure that we do not inadvertently pull a newer version of `poetry-core`.

Reply:
@finswimmer I made and merged the changes mentioned above on Poetry's side so we should be good to go for this PR.